File: src/type/github.py
from dataclasses import dataclass
from datetime import datetime
from typing import List, TypedDict


@dataclass
class Author(TypedDict):
    login: str
    id: int
    node_id: str
    avatar_url: str
    gravatar_id: str
    url: str
    html_url: str
    followers_url: str
    following_url: str
    gists_url: str
    starred_url: str
    subscriptions_url: str
    organizations_url: str
    repos_url: str
    events_url: str
    received_events_url: str
    type: str
    site_admin: bool


@dataclass
class Asset(TypedDict):
    url: str
    id: int
    node_id: str
    name: str
    label: None
    uploader: Author
    content_type: str
    state: str
    size: int
    download_count: int
    created_at: datetime
    updated_at: datetime
    browser_download_url: str


@dataclass
class IGithubLatestReleases(TypedDict):
    url: str
    assets_url: str
    upload_url: str
    html_url: str
    id: int
    author: Author
    node_id: str
    tag_name: str
    target_commitish: str
    name: str
    draft: bool
    prerelease: bool
    created_at: datetime
    published_at: datetime
    assets: List[Asset]
    tarball_url: str
    zipball_url: str
    body: str

File: types/github.go
package types

type GitHubLatestReleases struct {
	URL             string   `json:"url"`
	AssetsURL       string   `json:"assets_url"`
	UploadURL       string   `json:"upload_url"`
	HTMLURL         string   `json:"html_url"`
	ID              int64    `json:"id"`
	Author          GHAuthor `json:"author"`
	NodeID          string   `json:"node_id"`
	TagName         string   `json:"tag_name"`
	TargetCommitish string   `json:"target_commitish"`
	Name            string   `json:"name"`
	Draft           bool     `json:"draft"`
	Prerelease      bool     `json:"prerelease"`
	CreatedAt       string   `json:"created_at"`
	PublishedAt     string   `json:"published_at"`
	Assets          []Asset  `json:"assets"`
	TarballURL      string   `json:"tarball_url"`
	ZipballURL      string   `json:"zipball_url"`
	Body            string   `json:"body"`
}

type Asset struct {
	URL                string      `json:"url"`
	ID                 int64       `json:"id"`
	NodeID             string      `json:"node_id"`
	Name               string      `json:"name"`
	Label              interface{} `json:"label"`
	Uploader           Author      `json:"uploader"`
	ContentType        string      `json:"content_type"`
	State              string      `json:"state"`
	Size               int64       `json:"size"`
	DownloadCount      int64       `json:"download_count"`
	CreatedAt          string      `json:"created_at"`
	UpdatedAt          string      `json:"updated_at"`
	BrowserDownloadURL string      `json:"browser_download_url"`
}

type GHAuthor struct {
	Login             string `json:"login"`
	ID                int64  `json:"id"`
	NodeID            string `json:"node_id"`
	AvatarURL         string `json:"avatar_url"`
	GravatarID        string `json:"gravatar_id"`
	URL               string `json:"url"`
	HTMLURL           string `json:"html_url"`
	FollowersURL      string `json:"followers_url"`
	FollowingURL      string `json:"following_url"`
	GistsURL          string `json:"gists_url"`
	StarredURL        string `json:"starred_url"`
	SubscriptionsURL  string `json:"subscriptions_url"`
	OrganizationsURL  string `json:"organizations_url"`
	ReposURL          string `json:"repos_url"`
	EventsURL         string `json:"events_url"`
	ReceivedEventsURL string `json:"received_events_url"`
	Type              string `json:"type"`
	SiteAdmin         bool   `json:"site_admin"`
}
